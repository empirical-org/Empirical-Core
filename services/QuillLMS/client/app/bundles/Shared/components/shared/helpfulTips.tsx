import * as React from 'react';

import { expandIcon } from '../../index'

const lightbulbIcon = <img alt="Lightbulb icon for helpful tips" src={`${process.env.CDN_URL}/images/pages/activity_analysis/lightbulb.svg`} />

const TipSection = ({ toggleSection, index, openStates, section, }) => {
  function handleClick() { toggleSection(index)}

  if (section.isDisabled) {
    return (
      <div className="toggle-student-report-explanation scoring-explanation is-closed">
        <span />
        <h4>{section.headerText}</h4>
      </div>
    )
  }

  if (openStates[index]) {
    return (
      <button className="toggle-student-report-explanation is-open" key={index} onClick={handleClick} type="button">
        <img alt={expandIcon.alt} src={expandIcon.src} />
        <div>
          <h4>{section.headerText}</h4>
          <div className="body">{section.body}</div>
        </div>
      </button>
    )
  }

  return (
    <button className="toggle-student-report-explanation scoring-explanation is-closed" onClick={handleClick} type="button">
      <img alt={expandIcon.alt} src={expandIcon.src} />
      <h4>{section.headerText}</h4>
    </button>
  )

}

export const HelpfulTips = ({ sections, header }) => {
  const [openStates, setOpenStates] = React.useState(sections.map(() => false));

  function toggleSection(index) {
    const newOpenStates = [...openStates];
    newOpenStates[index] = !newOpenStates[index];
    setOpenStates(newOpenStates);
  };

  return (
    <div className="helpful-tips">
      <div className="helpful-tips-header">
        {lightbulbIcon}
        {header}
      </div>
      {sections.map((section, index) => (
        <TipSection
          index={index}
          key={section.headerText}
          openStates={openStates}
          section={section}
          toggleSection={toggleSection}
        />
      ))}
    </div>
  );
};
