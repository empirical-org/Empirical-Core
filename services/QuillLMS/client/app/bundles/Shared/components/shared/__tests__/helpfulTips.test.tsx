import * as React from "react";
import { render } from "@testing-library/react";

import { HelpfulTips, } from "../helpfulTips";

const props = {
  header: <h3>Helpful Tips <span>(Expand to show more information)</span></h3>,
  sections: [
    {
      headerText: "You will receive full credit on each question where you reach a strong response by the final attempt.",
      body: (
        <React.Fragment>
          <p>Questions where you reached a strong response by your final attempt will be indicated in green.</p><br />
          <p>Questions where you didn’t reach a strong response by your final attempt will be indicated in yellow.</p><br />
          <p>Your score is calculated by how many strong responses you got out of the total number of questions:</p>
          <ul>
            <li>Green for scoring between 83-100%</li>
            <li>Yellow for scoring between 32%-82%</li>
            <li>Red for scoring between 0-31%</li>
          </ul><br />
          <p>We encourage you to replay activities to get additional practice on skills and <a href="https://www.quill.org/teacher-center/go-for-green" rel="noopener noreferrer" target="_blank">Go for Green.</a></p>
        </React.Fragment>
      )
    },
    {
      headerText: "The bolded text helps you see edits you made to your writing.",
      body: (
        <React.Fragment>
          <p>In each of your responses, we have bolded all of the text that was added or edited from your previous response so that you can quickly see what you changed as you worked through the activity.</p><br />
          <p>This is different from the bolding you see within an activity, which provides hints on what to improve. In the feedback you see below, phrases like “look at the bolded word” refer to the bolding you got as a hint within the activity, not the bolded text you see on this report.</p>
        </React.Fragment>
      )
    },
    {
      headerText: 'This one is disabled',
      isDisabled: true
    }
  ]
}

describe('HelpfulTips', () => {
  test('it should render', () => {
    const { asFragment } = render(<HelpfulTips {...props} />);
    expect(asFragment()).toMatchSnapshot();
  })
})
