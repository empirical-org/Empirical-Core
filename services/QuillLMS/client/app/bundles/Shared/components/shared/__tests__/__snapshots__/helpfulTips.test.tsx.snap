// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpfulTips it should render 1`] = `
<DocumentFragment>
  <div
    class="helpful-tips"
  >
    <div
      class="helpful-tips-header"
    >
      <img
        alt="Lightbulb icon for helpful tips"
        src="undefined/images/pages/activity_analysis/lightbulb.svg"
      />
      <h3>
        Helpful Tips 
        <span>
          (Expand to show more information)
        </span>
      </h3>
    </div>
    <button
      class="toggle-student-report-explanation scoring-explanation is-closed"
      type="button"
    >
      <img
        alt="Expand icon"
        src="undefined/images/icons/expand.svg"
      />
      <h4>
        You will receive full credit on each question where you reach a strong response by the final attempt.
      </h4>
    </button>
    <button
      class="toggle-student-report-explanation scoring-explanation is-closed"
      type="button"
    >
      <img
        alt="Expand icon"
        src="undefined/images/icons/expand.svg"
      />
      <h4>
        The bolded text helps you see edits you made to your writing.
      </h4>
    </button>
    <div
      class="toggle-student-report-explanation scoring-explanation is-closed"
    >
      <span />
      <h4>
        This one is disabled
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
